plugins/org.eclipse.n4js/src/org/eclipse/n4js/n4JS/extensions/SourceElementExtensions.xtend

```diff
@@ -124,7 +139,17 @@ class SourceElementExtensions {
 			val next = allContents.next
 			switch (next) {
 				N4ClassDeclaration: {
-					next.collectVisibleTypedElement(addHere)
+					val polyfilledOrOriginalType = next.getTypeOrPolyfilledType();
+					val nonNullClassDecl = if (polyfilledOrOriginalType instanceof TClass) {
+						val n4ClassDecl = polyfilledOrOriginalType?.astElement as N4ClassDeclaration;
+						if (n4ClassDecl === null)
+							next
+						else
+							n4ClassDecl;
+					} else
+						next;
+
+					nonNullClassDecl.collectVisibleTypedElement(addHere)
```

Comment:
Could this be required for interfaces as well? (i.e. following case within this switch)

Reply:
Thanks for the hint. For the record: We looked into this and annotations of `@Polyfill` or `@StaticPolyfill` cannot be used for interfaces.